redeployContract.js

```diff
@@ -134,27 +137,32 @@ const portContract = async (contract_address,
     }
     
 
-    // TODO: Clean up this mess
-    let deploy_code = `0x${createB.createBytecode(contract_code, storage)}`;
-    // TODO: Should calculate proper gas requirement for executing code
-    console.log('Length', deploy_code.length);
-    if (deploy_code.length < code_size) {
-        return await deployLogic(target_web3, target_address, deploy_code);
+    // calculate how much the migration costs
+    const deploy_code = `0x${createB.createBytecode(contract_code, storage)}`;
+    const initContractInstance = new target_web3.eth.Contract([]);
+    const estimatedGas = await initContractInstance.deploy({ data: deploy_code }).estimateGas({
+        to: target_address,
+        gas: config.chain.gasLimit
+    });
+    console.log(`Estimated gas for migration: ${estimatedGas}`);
```

Comment:
In order to prevent overflows in the bytecode creation for very large key/value sets, I would propose to first estimate roughly if the gas limit will be exceeded before constructing the bytecode. To do so, we would require a config field storageCost (default 20,000). We could then first check with something like:
`if (Object.keys(storage).length * config.storageCost + config.chain.gasBuffer <= config.chain.gasLimit)`

If true, the precise estimated will be applied.

Reply:
Will do!